src/main/java/org/opensearch/knn/index/util/IndexHyperParametersUtil.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.Version;
+import org.opensearch.knn.index.KNNSettings;
+
+/**
+ * This class acts as an abstraction to get the default hyperparameter values for different parameters used in the
+ * Nearest Neighbor Algorithm across different version of Index.
+ */
+@Log4j2
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```

Comment:
Why do we need a no args constructor?

Reply:
This is to ensure that this class is not getting instantiated, as this has only static function. Its more of a best practice.